pkg/encoding/hex/hex.go

```diff
@@ -0,0 +1,41 @@
+package hex
```

Comment:
I don't see any go tests for this one like your other PRs is that intentional?

Also could we add a test for the failure case of `decode` please 🙏🏻

Reply:
not sure about gotest. The only gotest i found seems to be more related to the engine itself. 

Most of the wrapper are just testing the lua wrapper directly.

Thank you for the review, I added the test case for the failure scenario.